packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -665,14 +672,19 @@ function updateOffscreenComponent(
         if (enableCache) {
           // Save the cache pool so we can resume later.
           spawnedCachePool = getOffscreenDeferredCachePool();
-          // We don't need to push to the cache pool because we're about to
-          // bail out. There won't be a context mismatch because we only pop
-          // the cache pool if `updateQueue` is non-null.
         }
       } else {
         nextBaseLanes = renderLanes;
       }
 
+      if (enableCache) {
+        // push the cache pool even though we're going to bail out
+        // because otherwise there'd be a context mismatch
+        if (current !== null) {
+          pushSpawnedCachePool(workInProgress, null);
```

Comment:
Nit: Let's move this block right above `pushRenderLanes` since they are always called together